query_test.go

```diff
@@ -0,0 +1,90 @@
+package surrealdb
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/surrealdb/surrealdb.go/pkg/marshal"
+)
+
+// TestSurrealDBLocal runs a local SurrealDB instance by calling
+// `NewTestSurrealDB(t)`, which is a dedicated SurrealDB instance with no data.
+func TestSurrealDBLocal(t *testing.T) {
+	t.Parallel()
+	cases := map[string]struct {
+		schema              []string
+		dbInteraction       func(*testing.T, *DB)
+		keepServerUponError bool
+	}{
+		"first case": {
+			schema: []string{
+				"CREATE user:x SET name = 'xxxx'",
+				"CREATE user:y SET name = 'yyyy'",
+			},
+			dbInteraction: func(t *testing.T, db *DB) {
+				data, err := db.Select("user")
+				if err != nil {
+					t.Fatalf("failed to select, %v", err)
+				}
+				type dataholder struct {
+					ID   string `json:"id,omitempty"`
+					Name string `json:"name,omitempty"`
+				}
+				d, err := marshal.SmartUnmarshal[dataholder](data, nil)
+				if err != nil {
+					t.Fatal(err)
+				}
+				want := []dataholder{
+					{ID: "user:x", Name: "xxxx"},
+					{ID: "user:y", Name: "yyyy"},
+				}
+				if diff := cmp.Diff(want, d); diff != "" {
+					t.Errorf("Result mismatch (-want +got):\n%s", diff)
+				}
+			},
+		},
+		"second case": {
+			schema: []string{
+				"CREATE user:x SET name = 'xxxx'",
+				"CREATE user:y SET name = 'yyyy'",
+			},
+			dbInteraction: func(t *testing.T, db *DB) {
+			},
+		},
+		"third case": {
+			schema: []string{
+				"CREATE user:x SET name = 'xxxx'",
+				"CREATE user:y SET name = 'yyyy'",
+				"CREATE user:y SET name = 'yyyy'", // Should fail
```

Comment:
> We can, however, verify the error for this.

Yes, absolutely, this is the point I was going to add -- that would mean I need a separate test check for failure. At that point, I wouldn't be able to use the `Prepare` method above (because it runs `t.Fatal`), and would probably need to use `db.Query` instead.

I'll update the table definition accordingly.